src/main/java/com/github/matidominati/medicalclinic/service/PatientService.java

```diff
@@ -0,0 +1,75 @@
+package com.github.matidominati.medicalclinic.service;
+
+import com.github.matidominati.medicalclinic.model.Patient;
+import com.github.matidominati.medicalclinic.repository.PatientRepositoryImpl;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class PatientService {
+    private final List<Patient> patients;
+    private final PatientValidator patientValidator;
+    private final PatientRepositoryImpl patientRepository;
+
+    public List<Patient> getAllPatients() {
+        return patients;
+    }
+
+    public Patient getPatient(String email) {
+        Optional<Patient> patient = patientRepository.findPatientByEmail(email);
```

Comment:
patientRepository.findPatientByEmail(email).orElseThrow(() -> throw new RuntimeException("Patient with the provided email does not exist"));